aci/resource_aci_cloudlb.go

```diff
@@ -0,0 +1,570 @@
+					"network",
+				}, false),
+			},
+			"relation_cloud_rs_ldev_to_cloud_subnet": &schema.Schema{
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				Set:      schema.HashString,
+			},
+			"aaa_domain_dn": &schema.Schema{
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				Set:      schema.HashString,
+			},
+		}, GetNameAliasAttrSchema(), GetAnnotationAttrSchema()),
+	}
+}
+
+func mapCloudL4L7LoadBalancerAttrs(annotation, status string, d *schema.ResourceData) map[string]string {
+	return map[string]string{
+		"activeActive":                       d.Get("active_active").(string),
+		"allowAll":                           d.Get("allow_all").(string),
+		"annotation":                         annotation,
+		"autoScaling":                        d.Get("auto_scaling").(string),
+		"contextAware":                       d.Get("context_aware").(string),
+		"customRG":                           d.Get("custom_rg").(string),
+		"devtype":                            d.Get("device_type").(string),
+		"funcType":                           d.Get("function_type").(string),
+		"instanceCount":                      d.Get("instance_count").(string),
+		"isCopy":                             d.Get("is_copy").(string),
+		"isInstantiation":                    d.Get("is_instantiation").(string),
+		"isStaticIP":                         d.Get("is_static_ip").(string),
+		"l4L7DeviceApplicationSecurityGroup": d.Get("l4l7_device_application_security_group").(string),
+		"l4L7ThirdPartyDevice":               d.Get("l4l7_third_party_device").(string),
+		"managed":                            d.Get("managed").(string),
+		"maxInstanceCount":                   d.Get("max_instance_count").(string),
+		"minInstanceCount":                   d.Get("min_instance_count").(string),
+		"name":                               d.Get("name").(string),
+		"nameAlias":                          d.Get("name_alias").(string),
+		"nativeLBName":                       d.Get("native_lb_name").(string),
+		"packageModel":                       d.Get("package_model").(string),
+		"promMode":                           d.Get("prom_mode").(string),
+		"scheme":                             d.Get("scheme").(string),
+		"size":                               d.Get("size").(string),
+		"sku":                                d.Get("sku").(string),
+		"svcType":                            d.Get("service_type").(string),
+		"targetMode":                         d.Get("target_mode").(string),
+		"trunking":                           d.Get("trunking").(string),
+		"type":                               d.Get("cloud_l4l7_load_balancer_type").(string),
+		"Version":                            d.Get("version").(string),
+		"status":                             status,
+	}
+}
+
+func mapCloudRsLDevToCloudSubnetAttrs(annotation, status string, subnetTargetDnList []string) []interface{} {
```

Comment:
You can remove annotation and subnetTargetDnList, pass in d instead, this would create consistency across the function parameters.

Reply:
Discussed offline.